eclair-core/src/main/scala/fr/acinq/eclair/payment/send/MultiPartPaymentLifecycle.scala

```diff
@@ -249,7 +249,7 @@ class MultiPartPaymentLifecycle(nodeParams: NodeParams, cfg: SendPaymentConfig,
     if (cfg.recordPathFindingMetrics) {
       val fees = event match {
         case Left(_) => request.routeParams.getMaxFee(cfg.recipientAmount)
-        case Right(paymentSent) => paymentSent.feesPaid
+        case Right(paymentSent) => paymentSent.feesPaid + paymentSent.localFees
```

Comment:
This is incorrect because we want to record the local fees if, and only if, this is a trampoline relayed payment. The info is available in the the `upstream`, which can be either `Local` or `Trampoline`.

```suggestion
        case Right(paymentSent) => 
          val localFees = cfg.upstream match {
            case _: Upstream.Local => 0.msat // no local fees when we are the origin of the payment
            case _: Upstream.Trampoline =>
              // in case of a relayed payment, we need to take into account the fee of the first channels
              paymentSent.parts.collect {
                // NB: the route attribute will always be defined here
                case p@PartialPayment(_, _, _, _, Some(route), _) => route.head.fee(p.amountWithFees)
              }.sum
          }
          paymentSent.feesPaid + localFees
```

Reply:
Done